js/apps/account-ui/src/oid4vci/Oid4Vci.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+    Select,
+    SelectList,
+    SelectOption,
+    PageSectionVariants,
+    PageSection,
+    ActionList,
+    ActionListItem,
+    List,
+    ListItem,
+    MenuToggleElement,
+    MenuToggle
+} from '@patternfly/react-core';
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useAlerts } from "@keycloak/keycloak-ui-shared";
+import { usePromise } from "../utils/usePromise";
+import { Page } from "../components/page/Page";
+import { parseResponse } from "../api/parse-response";
+import { useEnvironment } from '../root/KeycloakContext';
+
+interface CredentialsIssuer {
+  credential_issuer: string;
+  credential_endpoint: string;
+  authorization_servers: string[]; 
+  credential_configurations_supported: Map<string, SupportedCredentialConfiguration>
+}
+
+interface SupportedCredentialConfiguration {
+  id: string,
+  format: string,
+  scope: string
+}
+
+type VCState = {
+  dropdownItems: string[],
+  selectOptions: Map<string, SupportedCredentialConfiguration>,
+  credentialIssuer?: CredentialsIssuer,
+  issuerDid: string,
+  qrCode: string,
+  isOpen: boolean,
+  offerQRVisible: boolean
+}
+
+export const Oid4Vci = () => {
+    const { addAlert, addError } = useAlerts();
+    const context = useEnvironment();
+   
+    const { t } = useTranslation();
+    const initialState: VCState = {
+      dropdownItems: [],
+      selectOptions: new Map<string, SupportedCredentialConfiguration>(),
+      issuerDid: "",
+      qrCode: "",
+      isOpen: false,
+      offerQRVisible: false
+    }
+    const initialSelected = t('verifiableCredentialsSelectionDefault') 
+
+    const [selected, setSelected] = useState<string>(initialSelected);
+    const [vcState, setState] = useState<VCState>(initialState);
+    const url:string = context.environment.authUrl
+    const realm = context.environment.realm
+    const wellKnownIssuer = url + "/realms/" + realm + "/.well-known/openid-credential-issuer"
+
+    usePromise(
+      (signal) =>
+        Promise.all([
+          getIssuer(wellKnownIssuer),
+        ]),
```

Comment:
There is no need to wrap it as it's only one function
```suggestion
          getIssuer(wellKnownIssuer),
```